public/js/length_visualisation.js

```diff
@@ -0,0 +1,339 @@
+        .attr('transform','translate('+this._margin.left+','+this._margin.top+')');
+    this.setupTooltip();
+    this.draw_visualisation();
+  }
+
+  fill_data() {
+    var data = [];
+    this._hits.map(function (d) {
+      data.push(d.length)
+    })
+    // for (var i in this._hits) {
+    //   data.push(hits[i].length);
+    // }
+    return data;
+  }
+
+  get_seq_type(algorithm) {
+    var SEQ_TYPES = {
+        blastn: {
+            query_seq_type:   'nucleic_acid',
+            subject_seq_type: 'nucleic_acid'
+        },
+        blastp: {
+            query_seq_type:   'amino_acid',
+            subject_seq_type: 'amino_acid'
+        },
+        blastx: {
+            query_seq_type:   'nucleic_acid',
+            subject_seq_type: 'amino_acid'
+        },
+        tblastx: {
+            query_seq_type:   'nucleic_acid',
+            subject_seq_type: 'nucleic_acid'
+        },
+        tblastn: {
+            query_seq_type:   'amino_acid',
+            subject_seq_type: 'nucleic_acid'
+        }
+    };
+    return SEQ_TYPES[algorithm];
+  }
+
+  setupTooltip() {
+    $('[data-toggle="tooltip"]').tooltip({
+      'placement': 'top',
+      'container': 'body',
+      'html': 'true',
+      'delay': 0,
+      'white-space': 'nowrap'
+    });
+  }
+
+  get_evalue(bin) {
+    var evals = [];
+    for (var i in bin) {
+      for (var j in this._hits) {
+        if (this._hits[j].length == bin[i]) {
+          evals.push(this._hits[j].evalue);
+        }
+      }
+    }
+    return evals.sort(function (a,b) {
+      return d3.min([a,b]);
+    });
+    // return evals.map(function (d) {
+    //   return d3.min([1e-5, d]);
+    // })
+  }
+
+  tick_formatter(seq_type) {
+    var ticks = this._x_scale.ticks();
+    var format = d3.format('.1f');
+    var prefix = d3.formatPrefix(ticks[ticks.length - 1]);
+    var suffixes = {amino_acid: 'aa', nucleic_acid: 'bp'};
+    return function (d) {
+      if (d === 0) { return d; }
+      if (_.indexOf(ticks,d) >= 0) {
+        if (suffixes[seq_type] == 'aa') {
+          return (d + ' ' + suffixes[seq_type])
+        } else {
+          return (format(prefix.scale(d)) + ' ' + prefix.symbol + suffixes[seq_type]);
+        }
+        // return (format(prefix.scale(d)) + prefix.symbol + ' ' + suffixes[seq_type]);
+      } else {
+        return ;
+      }
+    };
+  }
+
+  create_bins() {
+    var bins = d3.layout.histogram()
+        .range(this._x_scale.domain())
+        .bins(this._x_scale.ticks(50))
+        (this._data);
+    return bins;
+  }
+
+  prettify_evalue(evalue) {
+    var regex = /(\d+)\.?(\d+)?e?([+-]\d+)?/;
+    var match = regex.exec(evalue);
+    var base  = match[1] + '.' + match[2];
+    var power = match[3];
+    var prettyEvalue = parseFloat(base).toFixed(2);
+    var returnString = '<br>E value: ' + prettyEvalue;
+    if (power !== undefined) {
+        returnString +=  ' &times; 10<sup>' + power + '</sup>';
+    }
+    return returnString;
+  }
+
+  fill_data2() {
```

Comment:
This function is very unclear - including names like data2, reusing the same variable name in different contexts, a method called set_evalue that matches based on length.